### ISSUE TITLE
Wrong YouTube channel downloaded when grabbing geoblocked channel

### ISSUE THREAD

**Issue by ivan** [rg3/youtube-dl#10009]
## Please follow the guide below

- You will be asked some questions and requested to provide some information, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *issue* (like that [x])
- Use *Preview* tab to see how your issue will actually look like

---

### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2016.07.05*. If it's not read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [x] I've **verified** and **I assure** that I'm running youtube-dl **2016.07.05**

### Before submitting an *issue* make sure you have:
- [x] At least skimmed through [README](https://github.com/rg3/youtube-dl/blob/master/README.md) and **most notably** [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones

### What is the purpose of your *issue*?
- [x] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### The following sections concretize particular purposed issues, you can erase any section (the contents between triple ---) not applicable to your *issue*

---

### If the purpose of this *issue* is a *bug report*, *site support request* or you are not completely sure provide the full verbose output as follows:

Add `-v` flag to **your command line** you run youtube-dl with, copy the **whole** output and insert it here. It should look similar to one below (replace it with **your** log inserted between triple ```):

```
# PYTHONPATH=. python3 -m youtube_dl -v 'https://www.youtube.com/user/kananishinoSMEJ/videos'
[debug] System config: []
[debug] User config: []
[debug] Command-line args: ['-v', 'https://www.youtube.com/user/kananishinoSMEJ/videos']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2016.07.05
[debug] Git HEAD: 4080530
[debug] Python version 3.5.1+ - Linux-4.4.8-std-1-armv7l-with-Ubuntu-16.04-xenial
[debug] exe versions: ffmpeg 2.8.6-1ubuntu2, ffprobe 2.8.6-1ubuntu2, rtmpdump 2.4
[debug] Proxy map: {}
[youtube:user] kananishinoSMEJ: Downloading channel page
[youtube:playlist] UUWeg2Pkate69NFdBeuRFTAw: Downloading webpage
[download] Downloading playlist: Uploads from SQUEEZIE
[youtube:playlist] UUWeg2Pkate69NFdBeuRFTAw: Downloading page #1
[youtube:playlist] UUWeg2Pkate69NFdBeuRFTAw: Downloading page #2
[youtube:playlist] UUWeg2Pkate69NFdBeuRFTAw: Downloading page #3
[youtube:playlist] UUWeg2Pkate69NFdBeuRFTAw: Downloading page #4
^C
ERROR: Interrupted by user
```

```
# PYTHONPATH=. python3 -m youtube_dl -v 'https://www.youtube.com/user/kananishinoSMEJ'       
[debug] System config: []
[debug] User config: []
[debug] Command-line args: ['-v', 'https://www.youtube.com/user/kananishinoSMEJ']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2016.07.05
[debug] Git HEAD: 4080530
[debug] Python version 3.5.1+ - Linux-4.4.8-std-1-armv7l-with-Ubuntu-16.04-xenial
[debug] exe versions: ffmpeg 2.8.6-1ubuntu2, ffprobe 2.8.6-1ubuntu2, rtmpdump 2.4
[debug] Proxy map: {}
[youtube:user] kananishinoSMEJ: Downloading channel page
[youtube:playlist] UUWeg2Pkate69NFdBeuRFTAw: Downloading webpage
[download] Downloading playlist: Uploads from SQUEEZIE
[youtube:playlist] UUWeg2Pkate69NFdBeuRFTAw: Downloading page #1
[youtube:playlist] UUWeg2Pkate69NFdBeuRFTAw: Downloading page #2
[youtube:playlist] UUWeg2Pkate69NFdBeuRFTAw: Downloading page #3
[youtube:playlist] UUWeg2Pkate69NFdBeuRFTAw: Downloading page #4
[youtube:playlist] UUWeg2Pkate69NFdBeuRFTAw: Downloading page #5
^C
ERROR: Interrupted by user
```

---

### Description of your *issue*, suggested solution and other information

Some YouTube channels are fully geoblocked, e.g. https://www.youtube.com/user/kananishinoSMEJ is accessible only from Japanese IP blocks.  Downloading that channel with youtube-dl outside of Japan results in youtube-dl grabbing an unrelated channel.

Possibly related to https://github.com/rg3/youtube-dl/issues/5904

**Comment by ivan** [comment]
It looks like

```py
            channel_playlist_id = self._html_search_meta(
                'channelId', channel_page, 'channel id', default=None)
```

fails to find anything and it falls back to

```py
            if not channel_playlist_id:
                channel_playlist_id = self._search_regex(
                    r'data-(?:channel-external-|yt)id="([^"]+)"',
                    channel_page, 'channel id', default=None)
```

and extracts an unrelated channel ID.

**Comment by yan12125** [comment]
Yep. It finds channels under the section "Related channels"

**Comment by ivan** [comment]
I have an even-more-paranoid version of a fix that involves extracting both `<meta itemprop="channelId"` and `<link rel="alternate" type="application/rss+xml" title="RSS" href="https://www.youtube.com/feeds/videos.xml?channel_id=` and making sure they match, would you take a PR with that?

**Comment by yan12125** [comment]
Haven't investigated. The fallback part is exactly a fix for #5904. Maybe @dstftw knows more for why not use the ```video.xml``` approach.
